mne/decoding/ssd.py

```diff
@@ -172,17 +177,22 @@ def fit(self, X, y=None):
 
         cov_signal = _regularized_covariance(
             X_signal, reg=self.reg, method_params=self.cov_method_params,
-            rank=self.rank, info=self.info)
+            rank="full", info=self.info)
```

Comment:
if rank will always be "full" then it should be removed from the Parameters of the SSD class.

Reply:
Here `rank` is specified to be "full" to ensure the transformations of the covariance matrices are not performed when they are being computed, but rather after the fact in `_dimensionality_reduction`, to which `rank` is passed, so it is still in use.